core/arch/x86/emit_utils.c

```diff
@@ -1354,13 +1354,59 @@ append_restore_simd_reg(dcontext_t *dcontext, instrlist_t *ilist, bool absolute)
         int i;
         uint opcode = move_mm_reg_opcode(true /*align16*/, true /*align32*/);
         ASSERT(proc_has_feature(FEATURE_SSE));
-        for (i = 0; i < proc_num_simd_saved(); i++) {
+        instr_t *post_restore = NULL;
```

Comment:
I will add a comment that preserve_xmm_caller_saved() is always true if ZMM_ENABLED().

Reply:
Done.